metrics_sdk/test/integration/periodic_metric_reader_test.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::SDK do
+  describe '#periodic_metric_reader' do
+    before { reset_metrics_sdk }
+
+    it 'emits 2 metrics after 10 seconds' do
+      OpenTelemetry::SDK.configure
+
+      metric_exporter = OpenTelemetry::SDK::Metrics::Export::InMemoryMetricPullExporter.new
+      periodic_metric_reader = OpenTelemetry::SDK::Metrics::Export::PeriodicMetricReader.new(interval_millis: 5, timout_millis: 5, exporter: metric_exporter)
+
+      OpenTelemetry.meter_provider.add_metric_reader(periodic_metric_reader)
+
+      meter = OpenTelemetry.meter_provider.meter('test')
+      counter = meter.create_counter('counter', unit: 'smidgen', description: 'a small amount of something')
+
+      counter.add(1)
+      counter.add(2, attributes: { 'a' => 'b' })
+      counter.add(2, attributes: { 'a' => 'b' })
+      counter.add(3, attributes: { 'b' => 'c' })
+      counter.add(4, attributes: { 'd' => 'e' })
+
+      sleep(8)
```

Comment:
If you're interested in an alternative to calling `sleep`, [New Relic uses these methods to advance, freeze, and unfreeze time in our tests](https://github.com/newrelic/newrelic-ruby-agent/blob/7c770dc74dd7fcf4e93f87a71295bdf104ad30c2/test/agent_helper.rb#L581-L629). It allows us to get the benefits of advancing time without slowing down the tests. We mostly use the `Process`-related methods now.

Incorporating [timecop](https://github.com/travisjeffery/timecop) could be another option.

Reply:
Thanks for the example, and I have tried both new relic method and the timecop. 
Not sure If I setup them correctly (with timecop/the code block), it didn't change the test time.

P.S. I think it would be great to have [New Relic uses these methods to advance, freeze, and unfreeze time in our tests](https://github.com/newrelic/newrelic-ruby-agent/blob/7c770dc74dd7fcf4e93f87a71295bdf104ad30c2/test/agent_helper.rb#L581-L629) in test_helpers gem

Reply:
Thanks for giving it a try! Good idea about adding the methods. I created https://github.com/open-telemetry/opentelemetry-ruby/issues/1668 to track this.